dev/PushNotifications/externs.h

```diff
@@ -0,0 +1,8 @@
+﻿#pragma once
+#include "pch.h"
+#include "PushNotificationReceivedEventArgs.h"
+
+extern wil::unique_handle g_waitHandleForArgs;
+extern wil::critical_section g_lock;
```

Comment:
g_lock is a global variable? That's an awfully generic name...

If you're making a lock for PushNotifications, at a minimum stick it in a namespace

Alternatively, if you need access control or want a set of code accessing some globals but not others make a class (in a namespace) with only static attributes + methods e.g.

```
namespace Microsoft.PushNotifications
{
    class Globals
    {
    public:
        static wil::critical_section Lock() { return g_lock };
    private:
        static wil::critical_section g_lock;
    };
}
```

then using your lock replace

```
...g_lock...
```

with

```
...Microsoft::PushNotifications::PushGlobals::Lock()...
```

or some such.

I find 'extern' is rarely used as even if it's legal, it's too easy for globals to collide (especially when multiple parties party in the same module, as in Microsoft.ProjectReunion.dll) and adding symbols to the global namespace degrades maintainability, searchability, ...

Reply:
Agree that it's a very generic name and easy to collide with, but `g_lock` is not needed anymore, so just going to remove.